app/routes/_layout+/calendar.tsx

```diff
@@ -0,0 +1,162 @@
+  {
+    id: "booking-5",
+    name: "March Booking 1",
+    from: new Date("2024-03-05"),
+    to: new Date("2024-03-10"),
+    status: "CONFIRMED",
+    creatorId: "user-5",
+  },
+  {
+    id: "booking-6",
+    name: "March Booking 2",
+    from: new Date("2024-03-15"),
+    to: new Date("2024-03-20"),
+    status: "CONFIRMED",
+    creatorId: "user-6",
+  },
+  {
+    id: "booking-5",
+    name: "April Booking 1",
+    from: new Date("2024-04-05"),
+    to: new Date("2024-04-10"),
+    status: "CONFIRMED",
+    creatorId: "user-6",
+  },
+  {
+    id: "booking-6",
+    name: "April Booking 2",
+    from: new Date("2024-04-15"),
+    to: new Date("2024-04-20"),
+    status: "CONFIRMED",
+    creatorId: "user-7",
+  },
+];
+
+// As per the DB Model
+const dummyBookingsResponse = {
+  bookings: dummyBookings,
+  bookingCount: dummyBookings.length,
+  page: 1,
+  perPage: 2,
+  search: null,
+  organizationId: dummyOrganizationId,
+  userId: dummyUserId,
+};
+
+// Loader Function to Return Bookings Data
+export const loader = ({ request }: LoaderFunctionArgs) => {
+  const url = new URL(request.url);
+  const monthParam = url.searchParams.get("month");
+  const yearParam = url.searchParams.get("year");
+
+  const currentMonth = monthParam
+    ? parseInt(monthParam, 10) - 1
+    : new Date().getMonth() + 1;
+
+  const currentYear = yearParam
+    ? parseInt(yearParam, 10)
+    : new Date().getFullYear();
+
+  // Filter the bookings for the selected month and year
+  const filteredBookings = dummyBookings.filter(
+    (booking) =>
+      booking.from.getMonth() === currentMonth &&
+      booking.from.getFullYear() === currentYear
+  );
+
+  const header = {
+    title: `Calendar for ${currentMonth + 1}/${currentYear}`,
+  };
+
+  return json({ bookings: filteredBookings, header });
+};
+
+interface CalendarEvent {
+  title: string;
+  start: string;
+  end: string;
+}
+// Calendar Component
+const Calendar = () => {
+  const { bookings, header } = useLoaderData<typeof loader>();
+  const navigate = useNavigate();
+  const [calendarEvents, setCalendarEvents] = useState<CalendarEvent[]>([]);
+  useEffect(() => {
+    if (bookings && bookings.length > 0) {
+      const events = bookings.map((booking) => ({
+        title: booking.name,
+        start: booking.from,
+        end: booking.to,
+      }));
+      setCalendarEvents(events); // Set the new events
+    }
+  }, [bookings]);
+
+  const handleMonthChange = (info: any) => {
+    const newMonth = !(info.start.getDate() == 1)
+      ? info.start.getMonth() + 1
+      : info.start.getMonth();
+    const newYear = info.start.getFullYear();
+    navigate(`?month=${newMonth + 1}&year=${newYear}`);
+  };
+
+  return (
+    <>
+      <FullCalendar
+        plugins={[dayGridPlugin]}
+        firstDay={1}
+        initialView="dayGridMonth"
+        timeZone="local"
+        events={calendarEvents}
+        datesSet={handleMonthChange}
+      />
+    </>
+  );
+};
+
+const calendar = () => <Calendar />;
```

Comment:
Not sure why you do it like this. You can just directly export default Calendar

Reply:
yes, we can do that as well. Made the changes accordingly!